File: android/app/src/main/java/com/phicdy/totoanticipation/presenter/SettingPresenter.java
package com.phicdy.totoanticipation.presenter;

import android.support.annotation.NonNull;

import com.phicdy.totoanticipation.model.scheduler.DeadlineAlarm;
import com.phicdy.totoanticipation.model.storage.SettingStorage;
import com.phicdy.totoanticipation.view.SettingView;

import java.util.Date;


public class SettingPresenter implements Presenter {
    private final DeadlineAlarm alarm;
    private SettingView view;
    private Date deadline;
    private SettingStorage storage;

    public SettingPresenter(@NonNull DeadlineAlarm alarm, @NonNull Date deadline,
                            @NonNull SettingStorage storage) {
        this.alarm = alarm;
        this.deadline = deadline;
        this.storage = storage;
    }

    public void setView(SettingView view) {
        this.view = view;
    }

    @Override
    public void onCreate() {
    }

    @Override
    public void onResume() {
    }

    @Override
    public void onPause() {
    }

    public void activityCreate() {
        view.initView();
        view.initListener();
    }

    public void onLicenseClicked() {
        view.goToLicenseActivity();
    }

    public void onDeadlineNotificationSettingClicked(boolean checked) {
        storage.setDeadlineNotify(checked);
        if (checked) {
            if (deadline.getTime() == 0) return;
            alarm.set5hoursBefore(deadline);
        } else {
            alarm.cancel();
        }
    }
}

File: android/app/src/main/java/com/phicdy/totoanticipation/presenter/SettingPresenter.kt
package com.phicdy.totoanticipation.presenter

import com.phicdy.totoanticipation.legacy.model.storage.SettingStorage
import com.phicdy.totoanticipation.legacy.presenter.Presenter
import com.phicdy.totoanticipation.legacy.view.SettingView
import com.phicdy.totoanticipation.model.scheduler.DeadlineAlarm
import java.util.Date


class SettingPresenter(private val alarm: DeadlineAlarm, private val deadline: Date,
                       private val storage: SettingStorage) : Presenter {
    private lateinit var view: SettingView

    fun setView(view: SettingView) {
        this.view = view
    }

    override fun onCreate() {}

    override fun onResume() {}

    override fun onPause() {}

    fun activityCreate() {
        view.initView(storage.isDeadlineNotify)
        view.initListener()
    }

    fun onLicenseClicked() {
        view.goToLicenseActivity()
    }

    fun onDeadlineNotificationSettingClicked(checked: Boolean) {
        storage.isDeadlineNotify = checked
        if (checked) {
            if (deadline.time == 0L) return
            alarm.setAtNoonOf(deadline)
        } else {
            alarm.cancel()
        }
    }

    fun onPrivacyPolicyClicked() {
        view.openPrivacyPolicy()
    }
}
